byoeb-v1/byoeb/byoeb/background_jobs/did_you_know/send_dyk.py

```
@@ -299,8 +318,18 @@ async def main(sheet: DykFactSheet, user_types: List[str], batch_size: int) -> N
 
     sheet: DykFactSheet = {lang: {} for lang in lang_cols.keys()}
     for row in reader:
-        id = str(uuid.UUID(row[guid_col]))  # validate uuids, bail early if in invalid format
+        # Skip empty rows or rows without GUID
+        if len(row) <= guid_col or not row[guid_col].strip():
+            continue
+        try:
+            id = str(uuid.UUID(row[guid_col].strip()))  # validate uuids
+        except Exception:
+            run_logger.error("Invalid GUID in %s: %s", SOURCE_PATH.name, row[guid_col])
+            continue
         for lang, lang_col in lang_cols.items():
+            # Guard against missing columns in the row
+            if len(row) <= lang_col:
+                continue
```

The changes here can be reverted, I believe this was needed to evade the .csv corruption - but we shouldn't have safeguards so that parse errors stay visible (fail fast).